src/main/java/com/cloudbees/jenkins/GitHubPushTrigger.java

```diff
@@ -154,7 +162,11 @@ public void run() {
     /**
      * Returns the file that records the last/current polling activity.
      */
-    public File getLogFile() {
+    public File getLogFile() throws IOException {
+        if (null == job) {
+            throw new IOException("Job was null, unable to getLogFile");
```

Comment:
Should this not be detected earlier in the cycle, rather than throwing an exception?

Reply:
The entire null checking here was added because the syntax package check start wouldn't pass.  Previously this, would have just thrown an NPE.  I don't have an actual example where this would occur, and I'm not familiar enough with the plugin to hazard any guesses.  Would you want this to be fixed as a precondition for this to get merged?  Or would a TODO here suffice (as it is also no worse than the current master, maybe better with the additional error messages that could help illustrate the condition in the wild)

Reply:
Ah I see, back to https://github.com/jenkinsci/github-plugin/pull/209#discussion_r269091269. I think the best fix would be for the polling cycle to start off assigning `job` to a local variable. If null, exit right away—unconfigured, nothing to do. Otherwise you have a non-null reference which can be used without further checks within this method and passed to any helper methods it calls.

Reply:
I have updated this, I changed the signature for `getLogFile` to take a `@NonNull Job<?, ?>` (made it private) and updated a few references.  I didn't explicitly make a local `job` variable, since I could just do the single null check in the start of the `onPost`, which then lets me use the class's `job` as a non-null field in the rest of the code block fine, and revert some of the job-null checks that were added.

I added a `getLogFile` wrapper to maintain API compatibility (I have no idea who would use this!).  But I also didn't want to touch modifying the public API and any unintended consequences that would have.

Reply:
It looks like spotbugs hates me :).  I'll add the local `job` variable as suggested to avoid this.

```

[2019-04-15T16:50:14.606Z] [ERROR] hudson.triggers.Trigger.job could be null and is guaranteed to be dereferenced in com.cloudbees.jenkins.GitHubPushTrigger$1.run() [com.cloudbees.jenkins.GitHubPushTrigger$1, com.cloudbees.jenkins.GitHubPushTrigger$1, com.cloudbees.jenkins.GitHubPushTrigger$1] Dereferenced at GitHubPushTrigger.java:[line 157]Dereferenced at GitHubPushTrigger.java:[line 160]Known null at GitHubPushTrigger.java:[line 156] NP_GUARANTEED_DEREF

[2019-04-15T16:50:14.606Z] [ERROR] Null passed for non-null parameter of GitHubPushTrigger.getLogFileForJob(Job) in com.cloudbees.jenkins.GitHubPushTrigger$1.run() [com.cloudbees.jenkins.GitHubPushTrigger$1, com.cloudbees.jenkins.GitHubPushTrigger$1] Method invoked at GitHubPushTrigger.java:[line 150]Known null at GitHubPushTrigger.java:[line 150] NP_NULL_PARAM_DEREF

[2019-04-15T16:50:14.606Z] [ERROR] Null passed for non-null parameter of GitHubPushTrigger.getLogFileForJob(Job) in com.cloudbees.jenkins.GitHubPushTrigger$1.runPolling() [com.cloudbees.jenkins.GitHubPushTrigger$1, com.cloudbees.jenkins.GitHubPushTrigger$1] Method invoked at GitHubPushTrigger.java:[line 107]Known null at GitHubPushTrigger.java:[line 107] NP_NULL_PARAM_DEREF

[2019-04-15T16:50:14.606Z] [INFO] 

```

Reply:
That sounds easiest. Again these are just my suggestions as an outside observer of the plugin.

Reply:
These changes are already pushed/and CI came back green